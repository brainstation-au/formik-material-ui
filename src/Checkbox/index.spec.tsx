import { Formik } from 'formik';
import React from 'react';
import renderer from 'react-test-renderer';
import { Checkbox } from '.';

it('renders correctly', () => {
  type FormValues = { employed: boolean };
  const initialValue: FormValues = { employed: false };
  const promise = Promise.resolve();
  // eslint-disable-next-line @typescript-eslint/no-unused-vars
  const onSubmitMock = jest.fn((_) => promise);
  const formik = (
    <Formik
      initialValues={initialValue}
      onSubmit={(values) => onSubmitMock(values)}
    >
      <form>
        <Checkbox name="employed" label="Employment Status" />
      </form>
    </Formik>
  );
  const tree = renderer.create(formik).toJSON();
  expect(tree).toMatchSnapshot();
});
