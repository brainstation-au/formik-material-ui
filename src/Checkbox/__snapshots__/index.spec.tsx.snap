// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<form>
  <div
    className="MuiFormControl-root css-1nrlq1o-MuiFormControl-root"
  >
    <label
      className="MuiFormControlLabel-root MuiFormControlLabel-labelPlacementEnd css-j204z7-MuiFormControlLabel-root"
      onBlur={[Function]}
    >
      <span
        className="MuiCheckbox-root MuiCheckbox-colorPrimary MuiButtonBase-root MuiCheckbox-root MuiCheckbox-colorPrimary PrivateSwitchBase-root css-12wnr2w-MuiButtonBase-root-MuiCheckbox-root"
        onBlur={[Function]}
        onContextMenu={[Function]}
        onDragLeave={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        onKeyUp={[Function]}
        onMouseDown={[Function]}
        onMouseLeave={[Function]}
        onMouseUp={[Function]}
        onTouchEnd={[Function]}
        onTouchMove={[Function]}
        onTouchStart={[Function]}
        tabIndex={null}
      >
        <input
          checked={false}
          className="PrivateSwitchBase-input css-1m9pwf3"
          data-indeterminate={false}
          disabled={false}
          name="employed"
          onChange={[Function]}
          type="checkbox"
          value={true}
        />
        <svg
          aria-hidden={true}
          className="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
          data-testid="CheckBoxOutlineBlankIcon"
          focusable="false"
          viewBox="0 0 24 24"
        >
          <path
            d="M19 5v14H5V5h14m0-2H5c-1.1 0-2 .9-2 2v14c0 1.1.9 2 2 2h14c1.1 0 2-.9 2-2V5c0-1.1-.9-2-2-2z"
          />
        </svg>
      </span>
      <span
        className="MuiTypography-root MuiTypography-body1 MuiFormControlLabel-label css-ahj2mt-MuiTypography-root"
      >
        Employment Status
      </span>
    </label>
    <p
      className="MuiFormHelperText-root MuiFormHelperText-sizeMedium MuiFormHelperText-contained css-1wc848c-MuiFormHelperText-root"
    />
  </div>
</form>
`;
